src/common/input/DatePicker.tsx

```diff
@@ -0,0 +1,337 @@
+        filterDate={filterDate}
+        locale={fi}
+        dateFormat={'dd.MM.yyyy'}
+        showMonthDropdown={true}
+        peekNextMonth={true}
+        showYearDropdown={true}
+        dropdownMode={'select'}
+        startDate={value ? getDateObject(value) : null}
+        scrollableYearDropdown={true}
+        yearDropdownItemNumber={100}
+        minDate={getMinDate()}
+        maxDate={getMaxDate()}
+        dateFormatCalendar={DISPLAYED_FORMAT}
+        popperContainer={renderCalendarContainer} // using popperContainer doesn't hurt. It might prevent some overflow issues (calendar not completely visible)
+        fixedHeight={true}
+      />
+      <DatePickerWrapperStyles />
+    </DatePickerWrapper>
+  )
+})
+
+const renderCalendarContainer = ({
+  children,
+}: {
+  children: JSX.Element[]
+}): React.ReactNode => {
+  const el = document.getElementById('root')
+  return ReactDOM.createPortal(children, el!)
+}
+
+const renderDatePickerInput = ({
+  onInputChange,
+  isValidDate,
+  onInputBlur,
+  placeholder,
+  value,
+  label,
+  disabled,
+  isEmptyValueAllowed,
+  openCalendar,
+  attrs,
+}: {
+  onInputChange: (value: any) => void
+  isValidDate: (dateString: string) => boolean
+  onInputBlur: () => void
+  placeholder: string
+  value?: string
+  label?: string
+  disabled?: boolean
+  isEmptyValueAllowed?: boolean
+  openCalendar: Function
+  attrs: any
+}) => {
+  const onChange = (value: string) => {
+    onInputChange(value)
+  }
+
+  let isInputValid = _.isEmpty(value) ? isEmptyValueAllowed : isValidDate(value!)
+  return (
+    <InputContainer>
+      <Input
+        style={isInputValid ? undefined : { backgroundColor: 'var(--light-red)' }}
+        type="text"
+        label={label}
+        value={value ? value : ''}
+        placeholder={placeholder}
+        onClick={() => openCalendar()}
+        onChange={onChange}
+        onBlur={onInputBlur}
+        disabled={disabled}
+        {...attrs}
+      />
+      <CalendarIconWrapper>
+        <Calendar height="1.5rem" />
+      </CalendarIconWrapper>
+    </InputContainer>
+  )
+}
+
+function isDateTypeAllowed(date: Date, acceptableDayTypes?: AcceptableDayType[]) {
+  if (!acceptableDayTypes || acceptableDayTypes.length === 0) {
+    return true
+  }
+
+  return acceptableDayTypes.some((acceptableDayType: AcceptableDayType) => {
+    if (acceptableDayType === 'mo' && isMonday(date)) {
+      return true
+    } else if (acceptableDayType === 'tu' && isTuesday(date)) {
+      return true
+    } else if (acceptableDayType === 'we' && isWednesday(date)) {
+      return true
+    } else if (acceptableDayType === 'th' && isThursday(date)) {
+      return true
+    } else if (acceptableDayType === 'fr' && isFriday(date)) {
+      return true
+    } else if (acceptableDayType === 'sa' && isSaturday(date)) {
+      return true
+    } else if (acceptableDayType === 'su' && isSunday(date)) {
+      return true
+    }
+    return false
+  })
+}
```

Comment:
```suggestion
// Define in util/dayType.ts helper:

export const dayTypes = ['Ma', 'Ti', 'Ke', 'To', 'Pe', 'La', 'Su']

export function getDayTypeFromDate(date: string | Date): string {
  if (!date) {
    return ''
  }

  let dateObj = getDateObject(date)
  return dayTypes[getISODay(dateObj) - 1]
}

// Suggestion for this function:

function isDateTypeAllowed(date: Date, acceptableDayTypes?: AcceptableDayType[]) {
  if (!acceptableDayTypes || acceptableDayTypes.length === 0) {
    return true
  }

  let dayTypeOfDate = getDayTypeFromDate(date)
  return acceptableDayTypes.includes(dayTypeOfDate)
}
```

While this type of implementation is quite readable, it is very duplicated and can be written more tersely. This is my suggestion which is similar to a function in the server codebase.

Reply:
Refactored the function but didn't make an util for it (it would only save 1 line of code, I think it's ok now.

Reply:
Ok cool, util is indeed not really necessary unless there are other users of the function. The idea was to mirror the code in the server repo.